### ISSUE TITLE
Class & Function Selector Doesn't Work

### ISSUE THREAD

**Issue by mrclary** [spyder-ide/spyder#14101]
### What steps will reproduce the problem?

The Class & Function Selector does not work for files that reside in directories that have `__init__.py` files.

## Versions

* Spyder version: 4.2.0.dev0 5271e103f
* Python version: 3.7.9
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Darwin 19.6.0

### Dependencies

```

# Mandatory:
applaunchservices >=0.1.7      :  0.2.1 (OK)
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  1.6.0 (OK)
diff_match_patch >=20181111    :  20200713 (OK)
intervaltree >=3.0.2           :  3.1.0 (OK)
IPython >=4.0                  :  7.18.1 (OK)
jedi =0.17.2                   :  0.17.2 (OK)
nbconvert >=4.0                :  6.0.7 (OK)
numpydoc >=0.6.0               :  1.1.0 (OK)
parso =0.7.0                   :  0.7.0 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.5 (OK)
psutil >=5.3                   :  5.7.2 (OK)
pygments >=2.0                 :  2.7.1 (OK)
pylint >=1.0                   :  2.6.0 (OK)
pyls >=0.35.0;<1.0.0           :  test-selector-01 (OK)
pyls_black >=0.4.6             :  0.4.6 (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  1.0.1 (OK)
qtconsole >=4.7.7              :  4.7.7 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
rtree >=0.8.3                  :  0.9.4 (OK)
setuptools >=39.0.0            :  50.3.0.post20201006 (OK)
sphinx >=0.6.6                 :  3.2.1 (OK)
spyder_kernels >=1.9.4;<1.10.0 :  1.10.0.dev0 (OK)
three_merge >=0.1.1            :  0.1.1 (OK)
watchdog >=0.10.3              :  0.10.3 (OK)
zmq >=17                       :  19.0.2 (OK)

# Optional:
cython >=0.21                  :  0.29.21 (OK)
matplotlib >=2.0.0             :  3.3.1 (OK)
numpy >=1.7                    :  1.19.1 (OK)
pandas >=1.1.1                 :  1.1.3 (OK)
scipy >=0.17.0                 :  1.5.2 (OK)
sympy >=0.7.3                  :  1.6.2 (OK)
```

**Comment by mrclary** [comment]
Attached is a MWE and gif illustrating the issue. Notice that Class & Function Selector works for files in directories that *do not* have `__init__.py` files and does not work do have `__init__.py` files. Note also that command completion is unaffected.

[class_selector.zip](https://github.com/spyder-ide/spyder/files/5466050/class_selector.zip)

![class_selector](https://user-images.githubusercontent.com/9618975/97724997-dff33880-1a8a-11eb-88f9-ab57614dad32.gif)

**Comment by ccordoba12** [comment]
I think this is related to PR #14058. Could you checkout the last merge before that one (which would be commit a31205d58c54bfc77d9555ffee691ba4b304481f) and try again?

**Comment by mrclary** [comment]
Issue manifests beginning with 54178cc8f854d7180bdc40ef5a0abe80ee084b60.
Looks like it is caused by https://github.com/palantir/python-language-server/pull/879.
Also looks like this issue may already be resolved by https://github.com/palantir/python-language-server/pull/881

**Comment by mrclary** [comment]
I'll close this issue. If https://github.com/palantir/python-language-server/pull/881 doesn't resolve, I'll reopen

**Comment by ccordoba12** [comment]
> I'll close this issue. If palantir/python-language-server#881 doesn't resolve, I'll reopen

An update to our subrepo with this fix will be merged in PR #14019.
